python_modules/dagster/dagster/components/core/snapshot.py

```diff
@@ -47,17 +47,22 @@ def get_package_entry_snap(key: PluginObjectKey, obj: object) -> PluginObjectSna
     type_data = []
     owners = []
     tags = []
+    aliases = []
     if isinstance(obj, type) and issubclass(obj, Component):
         type_data.append(_get_component_type_data(obj))
         spec = obj.get_spec()
         owners = spec.owners
         tags = spec.tags
+        if obj.__module__ != key.namespace:
+            # add the module that defines the class as an alias
+            aliases.append(PluginObjectKey(namespace=obj.__module__, name=obj.__name__))
```

Comment:
I think we should rename this class to `RegistryKey` fwiw.

Reply:
I have a few of these renames I will push up